code/game/machinery/_machinery.dm

```diff
@@ -517,7 +517,7 @@
 	var/is_dextrous = FALSE
 	if(isanimal(user))
 		var/mob/living/simple_animal/user_as_animal = user
-		if (ISADVANCEDTOOLUSER(user_as_animal))
+		if (user_as_animal.dextrous || HAS_TRAIT(user_as_animal, TRAIT_CAN_USE_MACHINES))
```

Comment:
@JohnFulpWillard I meant using the nuke should be its own trait. I don't want even the chance for you to be giving carps any power over anything other than nukes.

Reply:
I was scared of snowflaking it too much, at least without being told to.

Reply:
turns out advancedtooluser is the trait that lets you use tgui menus at all so everything after the first commit didn't even work